File: pset1/mario/less/mario.c
#include <cs50.h>
#include <stdio.h>

// Draws a half-pyramid of a specified height

int main(void)
{
    int height, line_num, position;

    do
    {
        height = get_int("Height: ");
    }
    while (height <= 0 || height > 8);

    // Outer loop to walk through the rows
    for (line_num = 1; line_num <= height; line_num++)
    {
        /* Inner loop to draw the line of spaces and hashes
           amount of spaces in each row equals to the height of the pyramid less number of the current line
           amount of hashes equals to the row number (counting from 1)
        */
        for (position = 0; position >= 0 && position < height - line_num; position++)
        {
            printf(" ");
        }

        for (position = height - line_num; position < height; position++)
        {
            printf("#");
        }

        printf("\n");
    }

}

File: pset6/mario/less/mario.py
from cs50 import get_int

# Ask user for height (0 < height > 9)
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

# Printing spaces and hashes in one line of code, woo-hoo!
for row in range(height):
    print(" " * (height - row - 1) + "#" * (row + 1))
